samples/drivers/led_pwm/boards/disco_l475_iot1.overlay

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/pwm/pwm.h>
+
+/ {
+	/delete-node/ leds;
+
+	pwmleds: pwmleds {
+		compatible = "pwm-leds";
+		status = "okay";
+
+		green_led_1: green_led_1 {
+			pwms = <&pwm2 1 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
+			label = "green LD1";
+		};
+		green_led_2: green_led_2 {
+			pwms = <&pwm15 1 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
+			label = "green LD2";
+		};
+	};
+};
+
+&timers2 {
+	status = "okay";
+	st,prescaler = <10000>;
+	pwm2: pwm {
+		status = "okay";
+		/delete-property/ pinctrl-0;
+		pinctrl-0 = <&tim2_ch1_pa5>;
+	};
+};
+
+&timers15 {
+	status = "okay";
+	st,prescaler = <10000>;
+	pwm15: pwm {
+		status = "okay";
+		pinctrl-0 = <&tim15_ch1_pb14>; /* CN1 D10 */
+		pinctrl-names = "default";
+	};
+};
```

Comment:
Shouldn't you add this nodes into the board dts file, eventually with a disabled status, to avoid a conflict with gpio-led nodes ? I can see it is done like that for nucleo_f091rc. And in this overlay file we would just have to configure the status of the different nodes to switch from gpio-leds to pwm-leds.

Reply:
As a user I would expect to find all the devices defined in the board dts files. And I would probably miss this overlay files in the led-pwm sample. But obviously it is your call !